File: src/main/java/org/myfintech/payment/exception/ProblemDetailSchema.java
package org.myfintech.payment.exception;

import io.swagger.v3.oas.annotations.media.Schema;
import java.util.List;
import java.util.Map;

@Schema(name = "ProblemDetail", description = "Standard error structure")
public class ProblemDetailSchema {
    @Schema(example = "about:blank") public String type;
    @Schema(example = "Bad Request") public String title;
    @Schema(example = "400") public int status;
    @Schema(example = "Validation failed") public String detail;
    @Schema(example = "/api/v1/payments") public String instance;
    @Schema(description = "List of validation errors") public List<Map<String, String>> errors;
}

File: src/main/kotlin/org/myfintech/payment/exception/ProblemDetailSchema.kt
package org.myfintech.payment.exception

import io.swagger.v3.oas.annotations.media.Schema

@Schema(name = "ProblemDetail", description = "Standard error structure")
class ProblemDetailSchema {
    @Schema(example = "about:blank")
    var type: String? = null

    @Schema(example = "Bad Request")
    var title: String? = null

    @Schema(example = "400")
    var status: Int = 0

    @Schema(example = "Validation failed")
    var detail: String? = null

    @Schema(example = "/api/v1/payments")
    var instance: String? = null

    @Schema(description = "List of validation errors")
    var errors: MutableList<MutableMap<String?, String?>?>? = null
}
